packages/http/src/mocker/index.ts

```diff
@@ -41,20 +44,51 @@ const mock: IPrismComponents<IHttpOperation, IHttpRequest, IHttpResponse, IMockH
 
       return config;
     }),
-    chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
-    chain(result => assembleResponse(result, payloadGenerator))
+    Reader.chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
+    Reader.chain(result => assembleResponse(result, payloadGenerator)),
+    Reader.chain(response =>
+      withLogger(logger =>
+        pipe(
+          response,
+          Either.map(response => runCallbacks({ resource, request: input.data, response })(logger)),
+          Either.chain(() => response)
+        )
```

Comment:
I thought about the placement of callbacks on the beginning, my thoughts on that:
1. Moving to `core` is not a good idea, they are specific to `http`. They may make no sense for other types of specifications
2. Callbacks are part of mocking. They are not applicable to other existing commands (meaning proxy for now)

Reply:
Keep in mind that maybe callbacks (an outbound event triggered by an incoming event) will be the first step towards AsyncAPI support, so maybe some of it could be core. If that is just nonsense at this point dont worry we can refactor later, but keep it in mind.

Reply:
Callbacks have well-defined, single-point-of-contact. We can move them around when new requirements appear.

Reply:
Yeah for now we can leave it as it is since it's working properly anyway and not critical. However I also think this will be probably moved at a certain point out of this and put in the factory file. I'll just add a comment in the code.